src/domain/accounts/domain/src/entities/access_token.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+///////////////////////////////////////////////////////////////////////////////
+
+use chrono::{DateTime, Utc};
+use jsonwebtoken::TokenData;
+use opendatafabric::{AccountID, Multihash};
+use serde::{Deserialize, Serialize};
+use uuid::Uuid;
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct KamuAccessToken {
+    pub prefix: String,
+    pub id: Uuid,
+    pub random_bytes: [u8; 16],
+    pub random_bytes_hash: Multihash,
+    pub checksum: u32,
+    pub base32_token: String,
+    pub composed_token: String,
+}
```

Comment:
This entire struct seems to contain intermediate data used only during the generation.
The repository only takes `id` and `random_bytes_hash`:

```
        self.access_token_repository
            .create_access_token(&AccessToken {
                id: kamu_access_token.id,
                token_name: token_name.to_string(),
                token_hash: kamu_access_token
                    .random_bytes_hash
                    .digest()
                    .try_into()
                    .unwrap(),
                created_at: Utc::now(),
                revoked_at: None,
                account_id: account_id.clone(),
            })
            .await?;
```            

Future GraphQL API will only need the `composed_token`.

This structure might be neeed within the token service module for internal needs, but why would we add it to the public interface??

Reply:
It will be also used for new CLI command

Reply:
Also used to compare with hash stored in database